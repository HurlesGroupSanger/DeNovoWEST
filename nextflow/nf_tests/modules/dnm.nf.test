nextflow_process {

    name "DNM module"
    script "modules/dnm.nf"
    process "FILTER_DNM"

    test("filter_dnm") {

        when {

			params {
						dnm = "$baseDir/nf_tests/test_data/dnm/filter_dnm/input/DNM_b38_MANE.tsv"
						gene_list = "$baseDir/nf_tests/test_data/dnm/filter_dnm/input/gene_list.tsv"

					}
					process {
						"""
						input[0] = file(params.dnm)
						input[1] = file(params.gene_list)
						"""
					}
        }

        then {

            assert process.success

            // Test that the output matches the latest output
            assert snapshot(process.out).match()

			// Check that the output example is still up to date (Kept and discarded DNM files)
            assert path(process.out.get(0).get(0)).md5 == path("$baseDir/../examples/dnm/filter_dnm/output/kept_dnm.tsv").md5
            assert path(process.out.get(1).get(0)).md5 == path("$baseDir/../examples/dnm/filter_dnm/output/discarded_dnm.tsv").md5
            
        }
    }
}

