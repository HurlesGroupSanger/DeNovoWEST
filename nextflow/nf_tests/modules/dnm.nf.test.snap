{
    "filter_dnm": {
        "content": [
            {
                "0": [
                    "dnm_kept.tsv:md5,aa09d3b0cbbcdbc148887caa6222ce9b"
                ],
                "1": [
                    "dnm_discarded.tsv:md5,cb3264d7c6216a8c2333f403b47c9952"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2025-01-13T19:05:48.11717393"
    }
}